File: app/src/main/java/com/ytempest/wanandroid/base/activity/AbstractActivity.java
package com.ytempest.wanandroid.base.activity;

import android.content.Context;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.ytempest.layoutinjector.LayoutInjector;

import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * @author heqidu
 * @since 2020/6/20
 */
public abstract class AbstractActivity extends AppCompatActivity {

    private Unbinder mBind;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        LayoutInjector.inject(this);
        mBind = ButterKnife.bind(this);
        onViewCreated();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (mBind != null) {
            mBind.unbind();
        }
    }

    public Context getContext() {
        return this;
    }

    public AppCompatActivity getActivity() {
        return this;
    }

    protected abstract void onViewCreated();
}

File: app/src/main/java/com/ytempest/wanandroid/base/activity/AbstractActivity.kt
package com.ytempest.wanandroid.base.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.viewbinding.ViewBinding
import com.ytempest.wanandroid.binding.inflateViewBindingGeneric

/**
 * @author heqidu
 * @since 21-2-7
 */
abstract class AbstractActivity<VB : ViewBinding> : AppCompatActivity() {

    lateinit var binding: VB

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = inflateViewBindingGeneric(layoutInflater)
        setContentView(binding.root)
        onViewCreated()
    }


    abstract fun onViewCreated()
}